### PR TITLE
Feature/graph measures 1

### PR DESCRIPTION
**Pull request by iyerr3**
Note: This PR will have to be rebased after #148 is merged.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/114/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/115/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/121/

**Comment by iyerr3** [comment]
Push-forced after rebasing to master.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/123/

File: src/ports/postgres/modules/graph/measures.sql_in [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+                         'edge',        -- Edge table
+                         'src=src_id, dest=dest_id, weight=edge_weight',
+                                        -- Edge arguments (NULL means use default naming)
+                         'out_apsp');        -- Output table of shortest paths
+</pre>
+
+-# Compute the diameter measure for the graph:
+<pre class="syntax">
+DROP TABLE IF EXISTS out_diameter;
+SELECT madlib.graph_diameter('out_apsp', 'out_diameter');
```
Comment (orhankislal) [inline src/ports/postgres/modules/graph/measures.sql_in]: Fails on PG9.4:
```
SELECT graph_diameter('out_apsp', 'out_diameter');
psql:/tmp/madlib.pw7L2w/graph/test/measures.sql_in.tmp:87: ERROR:  spiexceptions.UndefinedObject: could not find array type for data type integer[]
CONTEXT:  Traceback (most recent call last):
  PL/Python function "graph_diameter", line 23, in <module>
    return measures.graph_diameter(**globals())
  PL/Python function "graph_diameter", line 310, in graph_apsp_measures
  PL/Python function "graph_diameter", line 273, in diameter
PL/Python function "graph_diameter"
```

File: src/ports/postgres/modules/graph/test/measures.sql_in [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              relative_error(inverse_avg_dist, 0.3043) < 1e-2 and
+              relative_error(sum_inverse_dist, 3.6833) < 1e-2 and
+              k_degree = 7,
+              'Incorrect value for closeness')
+FROM out_closeness
+WHERE src_id = 0;
+
+-- Compute the diameter measure for graph
+DROP TABLE IF EXISTS out_diameter;
+SELECT graph_diameter('out_apsp', 'out_diameter');
```
Comment (orhankislal) [inline src/ports/postgres/modules/graph/test/measures.sql_in]: Fails on PG9.4:
```
SELECT graph_diameter('out_apsp', 'out_diameter');
psql:/tmp/madlib.pw7L2w/graph/test/measures.sql_in.tmp:87: ERROR:  spiexceptions.UndefinedObject: could not find array type for data type integer[]
CONTEXT:  Traceback (most recent call last):
  PL/Python function "graph_diameter", line 23, in <module>
    return measures.graph_diameter(**globals())
  PL/Python function "graph_diameter", line 310, in graph_apsp_measures
  PL/Python function "graph_diameter", line 273, in diameter
PL/Python function "graph_diameter"
```

File: src/ports/postgres/modules/graph/measures.sql_in [hunk 3]
```diff
@@ -0,0 +1,820 @@
...
+
+-# Compute the average path length measure:
+<pre class="syntax">
+DROP TABLE IF EXISTS out_avg_path_length;
+SELECT madlib.graph_avg_path_length('out_apsp', 'out_avg_path_length');
+SELECT * FROM out_avg_path_length;
+</pre>
+<pre class="result">
+ avg_path_length
+------------------
```
Comment (orhankislal) [inline src/ports/postgres/modules/graph/measures.sql_in]: This causes the result block separated into two halves. I tried `\------------------` and that seems to fix the issue. It does not happen with any other result blocks

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/124/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/125/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/128/

**Comment by iyerr3** [comment]
Merged with 06788cc

**Pull request closed without merging**